dvc/commands/studio.py

```diff
@@ -0,0 +1,213 @@
+import argparse
+import logging
+import os
+
+from funcy import get_in
+
+from dvc.cli.command import CmdBase
+from dvc.cli.utils import append_doc_link, fix_subparsers
+
+logger = logging.getLogger(__name__)
+
+DEFAULT_SCOPES = "live,dvc_experiment,view_url,dql,download_model"
+AVAILABLE_SCOPES = ["live", "dvc_experiment", "view_url", "dql", "download_model"]
+
+
+class CmdStudioLogin(CmdBase):
+    def run(self):
+        from dvc_studio_client.auth import check_token_authorization
+
+        from dvc.env import DVC_STUDIO_URL
+        from dvc.repo.experiments.utils import gen_random_name
+        from dvc.ui import ui
+        from dvc.utils.studio import STUDIO_URL
+
+        name = self.args.name or gen_random_name()
+        hostname = self.args.hostname or os.environ.get(DVC_STUDIO_URL) or STUDIO_URL
+
+        try:
+            device_code, token_uri = self.initiate_authorization(name, hostname)
+        except ValueError as e:
+            ui.error_write(str(e))
+            return 1
+
+        access_token = check_token_authorization(uri=token_uri, device_code=device_code)
+        if not access_token:
+            ui.write(
+                "failed to authenticate: This 'device_code' has expired.(expired_token)"
+            )
+            return 1
```

Comment:
Can we move all of these to `dvc-studio-client` as well (make it a single API) and catch proper exception to notify users? I am not sure we need to deal with these details of opening webbrowsers, etc here. WDYT?